### ISSUE TITLE
Relax MPCD restrictions on boxes

### ISSUE THREAD

**Issue by mphoward** [glotzerlab/hoomd-blue#773]
## Description

MPCD currently requires that the box size be (1) orthorhombic, (2) constant, and (3) a multiple of the MPCD cell size. Only requirement (3) is strictly necessary for the algorithm to work right, so I will check over the code to see if it makes sense to support other (e.g., deformed) boxes. This is change is not necessarily breaking but may not be the behavior expected by the user, so I will look into it for v3.

**Comment by mphoward** [comment]
We will need to address this as part of our new work.

**Comment by mphoward** [comment]
Not stale

**Comment by mphoward** [comment]
@joaander I am going to do my best to get this branch to a point that we can open a PR onto `trunk-major` by Monday. It has a bunch of changes that will potentially cause merge conflicts when you finalize 5.0, so I want to get it in as soon as I can.

**Comment by joaander** [comment]
What planned changes to do you expect to conflict?

**Comment by mphoward** [comment]
This change will only touch MPCD code, but it touches *alot* of it so I think the main conflicts will come from any global search & replace incidental to other changes. And then of course, application of code formatting.

[feature/mpcd-triclinic-box](https://github.com/glotzerlab/hoomd-blue/tree/feature/mpcd-triclinic-box)

**Comment by joaander** [comment]
It looks like the majority of those changes are in C++ files, so the mostly likely cause of conflicts is #1938 which is already merged. The switch to ruff may cause conflicts in the `.py` files. I plan to switch to ruff as the last step before a 5.0 release. 

I will be sure to review your PR soon after you post it.

**Comment by mphoward** [comment]
> It looks like the majority of those changes are in C++ files

Yes, that's right—this is a big generalization of some of the underlying C++ code. Sounds like a plan, thanks!

**Comment by mphoward** [comment]
Done in #1950
